adoc/chapters/programming_interface.adoc

```diff
@@ -963,207 +963,237 @@ are handled by throwing synchronous SYCL exceptions.
+
+'''
+
+.[apidef]#platform::get_backend_info#
+[source,role=synopsis,id=api:platform-get-backend-info]
 ----
-template <typename Param> typename Param::return_type get_backend_info() const
+template <typename Param>
+typename Param::return_type get_backend_info() const
 ----
-   a@ Queries this SYCL [code]#platform# for <<backend>>-specific
-      information requested by the template parameter [code]#Param#.
-      The type alias [code]#Param::return_type# must be defined in
-      accordance with the <<backend>> specification.
-      Must throw an [code]#exception# with the [code]#errc::backend_mismatch#
-      error code if the <<backend>> that corresponds with [code]#Param# is different
-      from the <<backend>> that is associated with this [code]#platform#.
 
-a@
-[source]
+_Constraints:_ The [code]#Param# must be a backend information descriptor for
+the platform class.
+
+_Throws:_ An [code]#exception# with the [code]#errc::backend_mismatch# error
+code if the backend that corresponds with [code]#Param# is different from the
+backend that is associated with this platform.
+
+Each information descriptor specifies the return value and may also specify
+preconditions, additional exceptions that are thrown, etc.
+
+'''
+
+.[apidef]#platform::has#
+[source,role=synopsis,id=api:platform-has]
 ----
 bool has(aspect asp) const
 ----
-   a@ Returns true if all of the SYCL [code]#devices# associated with this
-      SYCL [code]#platform# have the given <<aspect>>. Returns [code]#false# if this SYCL [code]#platform# does not
-      contain any devices.
 
-a@
-[source]
+_Returns:_ The value [code]#true# if all of the devices associated with this
+platform have the given <<aspect>>.
+Returns the value [code]#false# if this platform does not contain any devices.
+
+'''
+
+.[apidef]#platform::has_extension#
+[source,role=synopsis,id=api:platform-has-extension]
 ----
 bool has_extension(const std::string& extension) const
 ----
-   a@ Deprecated, use [code]#has()# instead.
 
-Returns [code]#true# if this SYCL [code]#platform# supports the extension queried by the [code]#extension# parameter. A SYCL [code]#platform# can only support an extension if all associated SYCL [code]#devices# support that extension. Returns [code]#false# if this SYCL [code]#platform# does not contain any devices.
+Deprecated by SYCL 2020.
 
-a@
-[source]
+{note}Use [api]#platform::has# instead.
+{endnote}
+
+_Returns:_ The value [code]#true# if this platform supports the extension
+queried by the [code]#extension# parameter.
+A platform only supports an extension if all associated devices support that
+extension.
+Returns [code]#false# if this platform does not contain any devices.
+
+'''
+
+.[apidef]#platform::get_devices#
+[source,role=synopsis,id=api:platform-get-devices]
 ----
 std::vector<device>
-get_devices(info::device_type deviceType = info::device_type::all) const
+get_devices(info::device_type type = info::device_type::all) const
```

Comment:
I guess need to be change in `Static member function` Table 24 too (https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#_device_interface). Sorry if this table doesn't exist anymore with those change,

Reply:
Makes sense.  Done in 576f2741a0568b45b0543d24125e99e7dd1ec8a0.